netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,213 @@
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+          editorPane = new JEditorPane {
+            self =>
+
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(true)
+            setBackground(InterfaceColors.CODE_BACKGROUND) // Set the background color)
+            setForeground(InterfaceColors.DEFAULT_COLOR) //Set the font color
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+      }
+    } catch {
+      case e: Exception =>{
+        OptionDialog.showCustom(this, I18N.gui.get("error.dialog.unknown"), e.getMessage, null)
+      }
+    }
+  }
+
+
+  // Lazily initialize the Markdown parser and renderer (can be reused for multiple calls)
+  private lazy val markdownParser = Parser.builder().build()
+  private lazy val htmlRenderer = HtmlRenderer.builder().build()
+
+  private def formatJsonObjectList(jsonObjectList: List[JsonObject]): String = {
+    jsonObjectList.map { obj =>
+      // Convert fullText from Markdown to HTML
+      val fullTextHtml = htmlRenderer.render(markdownParser.parse(obj.fullText))
+
+      // Format title, date, and the converted HTML fullText
+      s"<h3>* ${I18N.gui.get("dialog.interface.update")}: ${obj.date} -- ${obj.title}</h3>" +
+        s"$fullTextHtml"
+    }.mkString("<html>", "", "</html>")
+  }
+
+  private def getJsonObjectHead: Option[String] = {
+      val jsonContent = fetchJsonFromUrl()
+      jsonObjectList = parseJsonToList(jsonContent)
+
+    // Return the title of the first item if jsonObjectList is not null and has elements
+    jsonObjectList match {
+      case Nil =>
+        println("jsonObjectList is empty or Nil; hiding NotificationBanner.")
+        this.setVisible(false) // Hide the NotificationBanner panel
+        None // Return None if jsonObjectList is null or empty
+      case list =>
+        this.setVisible(true) // Ensure the NotificationBanner is visible
+        list.headOption.map(_.title) // Return the title of the head element if available
+    }
+  }
+
+  private def isShowNeeded(): Boolean = {
```

Comment:
Empty-parens are only used in the naming of Scala methods to indicate the presence of side-effects.  This method should cause no side-effects.  However, this method *does* have side-effects, in that it fetches from the URL (which need only happen once, in the constructor, if not higher up in the object hierarchy).

Reply:
Would you be able to expand? I'm unclear how to proceed here